src/pytest_postgresql/executor.py

```diff
@@ -136,19 +136,24 @@ def init_directory(self):
         # remove old one if exists first.
         self.clean_directory()
         init_directory = (
-            self.executable, 'initdb',
-            '-o "--auth=trust --username=%s"' % self.user,
-            '-D %s' % self.datadir,
+            self.executable, 'initdb', '-D {}'.format(self.datadir)
         )
 
         if self.password:
-            with tempfile.NamedTemporaryFile() as password_file:
+            with tempfile.NamedTemporaryFile(mode='w') as password_file:
                 init_directory += (
-                    '--pwfile "%s"' % password_file.name,
+                    '-o "--auth=trust --username={} --pwfile={}"'.format(
+                        self.user, password_file.name
+                    ),
                 )
                 password_file.write(self.password)
+                # Without seek(0), PostgreSQL will se password file as empty
+                password_file.seek(0)
```

Comment:
flush, we don't need to read, so no need to seek

Reply:
Right, I changed that to `flush()`